### PR TITLE
Rebalances black and green core reactions also adds  three items to green cores (fug)

### PR DESCRIPTION
**Pull request by ghost**
well the green slime core had a bunch of medicine you could get if you had access to a chem dispenser which is required to do those reaction? Seemed kinda useless so i removed the medicine reactions from green and put DD, rezadone, and peridoxin to the black core
Black cores now give 5u nanobots instead of 1u combat or 1u medical
i'll probably add some unique xenobio items to the green core later like a slime heart that needs to be coded and sprited and slime tiles
I'll probably include that in another PR that adds reactions to pyrite and sepia as well as excluding xenos from gold cores but right now i want to see what people think about this
"when you code it" they said
Update: raincoat will be replaced by a special yellow raincoat
Update:  yellow raincoat replaced with slime coat that works like a wintercoat that looks like a yellow raincoat
image
fuck my life the raincoat will be hell to code BUT I GOTTA DO IT
image
TO DO:
NADA
:cl: 
 * rscadd: black cores now give DD, peridoxin, rezadone, and regular nanobots
 * rscadd: added slime tiles, works like phazon tiles to slime people and slows everything else. you can get five from injecting iron into green cores
 * rscadd: added slime hearts, deals toxins to you if you hold it and if you're a slime person it heals toxins. You get one from injecting blood into green cores
 * rscadd: Adds a raincoat that you can get from the green core by injecting water, it can hold temp guns and slime lamps and gives cold protection.
 * rscdel: green cores no longer give medicine

### CODE REVIEW

**Comment by ghost** [comment]
pls no bully

**Comment by carshalash** [comment]
What did he mean by this.

**Comment by ghost** [comment]
Oh yeah, both cores still give their slime toxins with plasma

**Comment by Yelier** [comment]
Why exclude xenos from the gold core?

**Comment by ghost** [comment]
Oh shit I forgot to mention, when I exclude xenos from Golden cores I'm gonna a put a xeno egg reaction for dark purple cores so you can raise xenos in the max containment pens

**Comment by ghost** [comment]
Fuck

**Comment by ghost** [comment]
Right now dark purples are only used for more plasma in case there are no miners and you somehow run out, even then you can get one mutation from purple and be completely fine with plasma since it's like 10 sheets? Might be 5

**Comment by Probe1** [comment]
.. okay why a raincoat?  That isn't right.  It shouts "I'm a placeholder".  Either don't touch it or get to the tiles and slime heart content that you intend to do instead of leaving us holding the bag until you do.

**Comment by ghost** [comment]
@Probe1 monster musume memes, the raincoat will stay but I think I'll get a special sprite for it

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -1643,32 +1574,62 @@
...
+/datum/chemical_reaction/slimerezadone
+	name = "Slime Rezadone"
+	id = "m_rezadone"
+	result = REZADONE
+	required_reagents = list(BLOOD = 5)
+	result_amount = 10
+	required_container = /obj/item/slime_extract/black
+	required_other = 1
+
+/datum/chemical_reaction/slimecasp/on_reaction(var/datum/reagents/holder)
```
Comment (ComicIronic) [inline code/modules/reagents/Chemistry-Recipes.dm]: Is this meant to be `slimerezadone`?

**Comment by ComicIronic** [comment]
What's the value of the reference?

**Comment by SonixApache** [comment]
slime tiles sound interesting

File: code/modules/reagents/Chemistry-Recipes.dm (hunk 1 above)
Comment (ghost) [reply code/modules/reagents/Chemistry-Recipes.dm]: I'm honestly not sure, first time CODING instead of deleting a few lines

**Comment by ghost** [comment]
Well I guess the coat can give some resistance to extinguishers so that it's not just cosmetic? It IS a raincoat after all

File: code/modules/reagents/Chemistry-Recipes.dm (hunk 1 above)
Comment (ghost) [reply code/modules/reagents/Chemistry-Recipes.dm]: Oh fuck nevermind i see it, will fix when can

**Comment by ghost** [comment]
oh no! i messed up my local branch

**Comment by ghost** [comment]
alright so that should have fixed the rezadone line that wasn't actually copy pasted from the orange reaction

**Comment by ghost** [comment]
i got a not shit slime tile sprite 
image
what it'll do is slow down anyone but slime people and will come from green cores at 5 tiles each reaction
just need to CODE IT which won't be too much time

**Comment by ghost** [comment]
and i managed to fuck my local branch again

**Comment by ghost** [comment]
i manged to unfuck my local branch again

**Comment by ghost** [comment]
tiles soon(within the next hour)

**Comment by ghost** [comment]
DONE
now to the slime heart and coat

**Comment by ghost** [comment]
oh yeah they delete when broken, slime completely loses it's form if you blow it up

**Comment by ghost** [comment]
broken as in from bombs, you can crowbar them up and shit

**Comment by ghost** [comment]
thanks shadowmech

**Comment by N3X15** [comment]
You do realize you can edit posts, right?

**Comment by PJB3005** [comment]
2 tiles per core seems a bit low, make it like 4 or 6.

**Comment by ghost** [comment]
Yeah I know I can edit posts, I made it low because the slow is really strong but alright fam
HOPEFULLY i should get the slime heart code done by tomorrow?

**Comment by ghost** [comment]
slime hearts will heal 3 toxins each second when held by a slime person and deals 5 toxins if you're not a master race. It's 5/sec because why the hell would you be holding it?
you get one from injecting green cores with blood, this item is mostly for accidentally spraying yourself
Yes you have to actually have it in-hand for it to work god damn it took a long time to find that thing

**Comment by SonixApache** [comment]
It's a hand-held object and not an organ?

**Comment by ghost** [comment]
Yeah, it might be used as an organ when it's time to balance slime people into a round start race ;-)
Right now it's a good item that deals with the problem of going too fast and walking into your own extinguisher sprays, I kept it a simple resprited green core like the yellow power cell

File: code/game/objects/items/slime_heart.dm [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/obj/item/slime_heart
+	name = "Slime Heart"
```
Comment (PJB3005) [inline code/game/objects/items/slime_heart.dm]: Remove capitalization here, if it's capitalized BYOND treats it as a proper noun.

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 3]
```diff
@@ -142,4 +142,23 @@
 	siemens_coefficient = 1
 	max_amount = 60
 
-	material = "fabric"
\ No newline at end of file
+	material = "fabric"
+
+obj/item/stack/tile/slime
+	name = "Tile of slime"
```
Comment (PJB3005) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: Remove capitalization.

File: code/game/objects/items/slime_heart.dm [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+/obj/item/slime_heart/process()
+	if(iscarbon(loc) && !cd)
+		var/mob/living/carbon/C = loc
+		if(C.is_holding_item(src))
+			if(C.dna && C.dna.mutantrace == "slime")
+				C.adjustToxLoss(-3)
+			else
+				C.adjustToxLoss(5)
+			cd = 1
+			spawn(1 SECONDS)
```
Comment (PJB3005) [inline code/game/objects/items/slime_heart.dm]: Process gets called every 2 seconds so this shouldn't be needed.

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 5]
```diff
@@ -142,4 +142,23 @@
...
+	desc = "A flat piece of slime made through xenobiology"
+	icon_state = "tile-slime"
+	w_class = W_CLASS_MEDIUM
+	force = 1.0
```
Comment (PJB3005) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: Remove `.0` here.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/slime_heart.dm (hunk 4 above)
Comment (ghost) [reply code/game/objects/items/slime_heart.dm]: Which line exactly?

File: code/game/objects/items/slime_heart.dm (hunk 2 above)
Comment (ghost) [reply code/game/objects/items/slime_heart.dm]: Th-thanks byond

File: code/game/objects/items/slime_heart.dm (hunk 4 above)
Comment (PJB3005) [reply code/game/objects/items/slime_heart.dm]: Just the entire `cd` stuff.

**Comment by ghost** [comment]
oh yeah and removed the .0
i guess you wanted me to remove the cd stuff because you thought it's not working how i wanted it to or something since we spoke a bit on discord? It works as intended, ticks every second

File: code/game/objects/items/slime_heart.dm [hunk 6]
```diff
@@ -0,0 +1,28 @@
+/obj/item/slime_heart
+	name = "slime Heart"
```
Comment (Exxion) [inline code/game/objects/items/slime_heart.dm]: See now this just looks weird

Comment (ghost) [reply code/game/objects/items/slime_heart.dm]: fuck i forgot a letter

**Comment by ghost** [comment]
anything else?
probably gonna start that raincoat tomorrow
i'll slap it in with the winter coats so that it's not just a meme cosmetic

**Comment by ghost** [comment]
oh shit is adding a slimecoat this easy? i hope not

**Comment by ghost** [comment]
(lol

**Comment by ghost** [comment]
one more thing

**Comment by ghost** [comment]
noice

**Comment by ghost** [comment]
that SHOULD be it

File: code/game/objects/items/slime_heart.dm [hunk 7]
```diff
@@ -5,24 +5,17 @@
 	icon_state = "Slime-Heart"
 	item_state = "Slime-Heart"
 	w_class = W_CLASS_TINY
-	var/cd = 0
 
 /obj/item/slime_heart/New()
 	processing_objects.Add(src)
 	..()
 
 /obj/item/slime_heart/process()
```
Comment (PJB3005) [inline code/game/objects/items/slime_heart.dm]: Change this to `OnMobLife(var/mob)` and remove the `New` code

**Comment by Commandersand** [comment]
hmm i think I will add the broken galoshes while I wait

File: code/game/objects/items/slime_heart.dm [hunk 8]
```diff
@@ -0,0 +1,20 @@
+/obj/item/slime_heart
+	name = "slime heart"
+	desc = "You devil..."
+	icon = 'icons/mob/slimes.dmi'
+	icon_state = "Slime-Heart"
+	item_state = "Slime-Heart"
+	w_class = W_CLASS_TINY
+
+/obj/item/slime_heart/New()
```
Comment (PJB3005) [inline code/game/objects/items/slime_heart.dm]: You're not removing the object from `processing_objects`.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/slime_heart.dm (hunk 7 above)
Comment (ghost) [reply code/game/objects/items/slime_heart.dm]: dang c*ders

**Comment by ghost** [comment]
wait what

File: code/game/objects/items/slime_heart.dm (hunk 7 above)
Comment (ghost) [reply code/game/objects/items/slime_heart.dm]: wait isn't that the old code?

Comment (ghost) [reply code/game/objects/items/slime_heart.dm]: oh nvm

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**